sentry-android-gradle-plugin/src/main/groovy/io/sentry/android/gradle/SentryPlugin.groovy

```diff
@@ -204,23 +204,36 @@ class SentryPlugin implements Plugin<Project> {
 
                         def variantName = variant.buildType.name
                         def flavorName = variant.flavorName
+                        // When flavor is used in combination with dimensions, variant.flavorName will be a concatenation
+                        // of flavors of different dimensions
                         def propName = "sentry.properties"
-                        def possibleProps = [
-                                "${project.projectDir}/src/${variantName}/${propName}",
+                        // current flavor name takes priority
+                        def possibleProps = []
+                        variant.productFlavors.each {
+                            // flavors used with dimension come in second
+                            possibleProps.push("${project.projectDir}/src/${it.name}/${propName}")
+                        }
+
+                        possibleProps = [
                                 "${project.projectDir}/src/${flavorName}/${propName}",
+                                "${project.rootDir.toPath()}/src/${flavorName}/${propName}",
+                        ] + possibleProps + [
+                                "${project.projectDir}/src/${variantName}/${propName}",
```

Comment:
I think the `variantName` stuff should go before the product flavors.

Reply:
fair. Didn't want to rename it either in the same PR but that's actually the `buildType` as you know